src/include/kdb.h.in

```diff
@@ -134,80 +125,62 @@ int kdbSet (KDB *handle, KeySet *returned,
  **************************************/
 
 /* Basic Methods */
-Key *keyNew (const char *keyname, ...) ELEKTRA_SENTINEL;
-Key *keyVNew (const char *keyname, va_list ap);
-
-Key * keyCopy (Key *dest, const Key *source, elektraCopyFlags flags);
-
-int keyClear (Key *key);
-int keyDel (Key *key);
-
-uint16_t keyIncRef (Key *key);
-uint16_t keyDecRef (Key *key);
-uint16_t keyGetRef (const Key *key);
+ElektraKey * elektraKeyNew (const char * keyname, ...) ELEKTRA_SENTINEL;
+ElektraKey * elektraKeyVNew (const char * keyname, va_list ap);
 
-/* Meta Info */
-int keyRewindMeta (Key *key);
-const Key *keyNextMeta (Key *key);
-const Key *keyCurrentMeta (const Key *key);
+ElektraKey * elektraKeyCopy (ElektraKey * dest, const ElektraKey * source, elektraCopyFlags flags);
 
-int keyCopyMeta (Key *dest, const Key *source, const char *metaName);
-int keyCopyAllMeta (Key *dest, const Key *source);
+int elektraKeyClear (ElektraKey * key);
```

Comment:
```suggestion
```

should be moved to operations lib (see #4245)

**Note:** This is different from `elektraKeyCopy (key, NULL, ELEKTRA_KEY_CP_ALL)`. The intention is that this snippet

```c
elektraKeyClear (k);
```

is the same as this 

```c
if (elektraKeyDel (k) == 0) {
    k = elektraKeyNew ("/", ELEKTRA_KEY_END);
}
```

but without the new allocation. This could be useful in (some) loops to avoid repeated re-allocation. `elektraKeyCopy` would only clear name, value and metadata and also respects locks.